Allow literal zero string as sql expression

marcelto opened the pull request:
A sql expression cannot currently be a literal zero (useful in certain unions) since setExpression will never be called.

$select = new Select('foo');
$select->columns([
   'bar' => new Expression('0') //fails, '0' is never set as the expression and the default empty string is used instead resulting in malformed sql
]);

marcelto committed:
Allow literal zero string as sql expression

A sql expression cannot currently be a literal zero (useful in certain unions) since setExpression will never be called.

$select = new Select('foo');
$select->columns([
   'bar' => new Expression('0') //fails, '0' is never set as the expression and the default empty string is used instead resulting in malformed sql
]);

malukenho commented:
Missing unit tests and fix build.

weierophinney commented:
@malukenho We have errors on the master branch right now due to a change in the Travis environment that causes 5.5 and 5.6 to error for `Zend\Cache`; the build errors are not related to this change.

However, @marcelto : we **do** need unit tests for this before I can accept it to merge.

malukenho commented:
thanks for the explanation @weierophinney

marcelto committed:
Literal zero test case

tests that the constructor will set the expression property even when
the expression is the string '0'

marcelto commented:
added new test case

weierophinney commented:
Merged to develop for release with 2.4.
